packages/side-nav/src/SideNav.spec.tsx

```diff
@@ -1,300 +1,52 @@
 import React from 'react';
-import { render, cleanup, screen, fireEvent } from '@testing-library/react';
+import { render, cleanup, fireEvent } from '@testing-library/react';
 import { SideNav, SideNavGroup, SideNavItem } from '.';
-import { SideNavItemProps } from './SideNavItem';
-import { SideNavGroupProps } from './SideNavGroup';
 
 type renderedElement = HTMLElement | null;
 
 interface RenderedElements {
+  containerEl?: renderedElement;
   navEl?: renderedElement;
   groupEl?: renderedElement;
   headerContentEl?: renderedElement;
   defaultHeaderEl?: renderedElement;
   itemEl?: renderedElement;
   childEl?: renderedElement;
+  collapseToggle?: renderedElement;
 }
 
 describe('packages/side-nav', () => {
   const testIds = {
-    sideNav: 'side-nav',
+    sideNavContainer: 'side-nav-container',
     sideNavGroup: 'side-nav-group',
     sideNavHeader: 'side-nav-header',
     sideNavItem: 'side-nav-item',
     sideNavLink: 'side-nav-link',
+    collapseToggle: 'side-nav-collapse-toggle',
   };
-  let renderedEls: RenderedElements = {};
 
   const className = 'test-class-name';
-  const headerText = 'test-header-text';
-  const headerContent = (
-    <div data-testid={testIds.sideNavHeader}>Header As Content</div>
-  );
+
+  let renderedEls: RenderedElements = {};
```

Comment:
We should add a test for collapsing/expanding the SideNav with a key press

Reply:
I really tried but unfortunately there's not a great way to do this without something very hacky. As a workaround, I tested that the states update appropriately on the collapse toggle itself. For the navigation itself, the only thing that changes between states are the classNames for the display.

Reply:
if we're already using data-testids here, maybe we do something like data-testid="side-nav-${isOpen ? "open" : "closed"}"

Reply:
the naming can definitely be better in my example

Reply:
That doesn't test that the state is correct any more than testing that the aria-expanded prop on the button is correct.

Reply:
yeah 🤦